File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/SeatAssigning/Events/SeatAssigned.cs
using System;
using ENode.Eventing;

namespace Registration.SeatAssigning
{
    public class SeatAssigned : DomainEvent<Guid>
    {
        public int Position { get; private set; }
        public SeatType Seat { get; private set; }
        public Attendee Attendee { get; private set; }

        public SeatAssigned() { }
        public SeatAssigned(int position, SeatType seat, Attendee attendee)
        {
            Position = position;
            Seat = seat;
            Attendee = attendee;
        }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/domain/seatassigning/Events/SeatAssigned.java
package com.microsoft.conference.registration.domain.seatassigning.Events;

import com.microsoft.conference.registration.domain.SeatType;
import com.microsoft.conference.registration.domain.seatassigning.Models.Attendee;
import org.enodeframework.eventing.DomainEvent;

public class SeatAssigned extends DomainEvent<String> {
    public int position;
    public SeatType seatType;
    public Attendee attendee;

    public SeatAssigned() {
    }

    public SeatAssigned(int position, SeatType seatType, Attendee attendee) {
        this.position = position;
        this.seatType = seatType;
        this.attendee = attendee;
    }
}
